File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.js
function solution(l, r) {
    const answer = [];

    // l 이상 r 이하 모든 정수 반복
    for (let i = l; i <= r; i++) {
        const str = i.toString(); // 숫자를 문자열로 변환
        let isValid = true; // 조건에 맞는지 여부

        // 문자열의 각 자리수를 검사
        for (let ch of str) {
            if (ch !== '0' && ch !== '5') {
                isValid = false; // 0이나 5가 아니면 조건 불충족
                break;
            }
        }

        // 조건 만족하면 결과 배열에 추가
        if (isValid) {
            answer.push(i);
        }
    }

    // 결과가 비어있으면 [-1] 반환
    return answer.length === 0 ? [-1] : answer;
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.java
import java.util.*;

class Solution {
    public int[] solution(int l, int r) {
        List<Integer> temp = new ArrayList<>();

        // l부터 r까지 반복
        for (int i = l; i <= r; i++) {
            if (isOnlyZeroAndFive(i)) {
                temp.add(i); // 조건 만족 시 리스트에 추가
            }
        }

        // 조건 만족하는 숫자가 없다면 -1 반환
        if (temp.isEmpty()) {
            return new int[]{-1};
        }

        // 리스트를 배열로 변환
        int[] answer = new int[temp.size()];
        for (int i = 0; i < temp.size(); i++) {
            answer[i] = temp.get(i);
        }

        return answer;
    }

    // 숫자가 0과 5로만 이루어져 있는지 검사하는 메서드
    private boolean isOnlyZeroAndFive(int num) {
        String str = String.valueOf(num); // 숫자를 문자열로 변환
        for (char ch : str.toCharArray()) {
            if (ch != '0' && ch != '5') {
                return false; // 0과 5 이외의 숫자가 포함되면 false
            }
        }
        return true;
    }
}
